### PR TITLE
[TensorFlow][Inference]Multi-gpu support for TF inference

### PR DESCRIPTION
**Pull request by waytrue17**
*GitHub Issue #, if available:*
Multi-gpu support for TF inference. 
- Previously all serving workers are assigned to a single gpu device. This pr assigns workers to multiple gpu diverse to optimize gpu utilization.
- No test needs to be added as this pr does not change any serving functionality but gpu utilization. 
- The same change has been merged into the TF-inference-toolkit repo: https://github.com/aws/sagemaker-tensorflow-serving-container/pull/227

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [x] Revision A: `sagemaker_remote_tests = "standard"`
- [x] Revision B: `sagemaker_remote_tests = "rc"`
- [x] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [x] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ashishgupta023** [comment]
LGTM

**Comment by tejaschumbalkar** [comment]
LGTM. Lets revert the temporary configuration.

**Review by tejaschumbalkar — APPROVED** [review]

**Comment by tejaschumbalkar** [comment]
@waytrue17  The `test_batch_transform` fails on job due to timeout of 600 sec. Actual job on SageMaker takes 11 minutes to complete. Can we increase the timer?

**Comment by tejaschumbalkar** [comment]
> @waytrue17 The `test_batch_transform` fails on job due to timeout of 600 sec. Actual job on SageMaker takes 11 minutes to complete. Can we increase the timer?

Please address this comment as well

**Comment by waytrue17** [comment]
> @waytrue17 The `test_batch_transform` fails on job due to timeout of 600 sec. Actual job on SageMaker takes 11 minutes to complete. Can we increase the timer?

I do not think this PR would affect the timeoue. It's more likely to be a flaky issue. I re-ran the efa test and now it's passing the test: https://tiny.amazon.com/37r23fp4/IsenLink

**Comment by waytrue17** [comment]
Reverted the temp config since the efa test passed. Please proceed with release @tejaschumbalkar

File: dlc_developer_config.toml [hunk 1]
```diff
@@ -22,7 +22,7 @@ benchmark_mode = false
 
 [build]
 # Frameworks for which you want to disable both builds and tests
-# available frameworks - ["autogluon", "huggingface_tensorflow", "huggingface_pytorch", "huggingface_tensorflow_trcomp", "huggingface_pytorch_trcomp", "tensorflow", "mxnet", "pytorch"]
+# available frameworks - []
 skip_frameworks = ["autogluon", "huggingface_tensorflow", "huggingface_pytorch", "huggingface_tensorflow_trcomp", "huggingface_pytorch_trcomp", "mxnet", "pytorch"]
```
Comment (RadhikaB-97) [inline dlc_developer_config.toml:26]: Make the skip_frameworks list empty.

Comment (waytrue17) [reply dlc_developer_config.toml]: Done

**Review by RadhikaB-97 — APPROVED** [review]

**Pull request merged**